src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/Configuration/McpFunctionMetadataProvider.cs

```diff
@@ -121,16 +124,42 @@ private static bool TryGetPropertiesFromAttributes(IFunctionMetadata functionMet
         }
 
         var properties = new List<ToolProperty>();
+
         foreach (var parameter in method.GetParameters())
         {
             var toolAttribute = parameter.GetCustomAttribute<McpToolPropertyAttribute>();
-
-            if (toolAttribute is null)
+            if (toolAttribute is not null)
             {
+                properties.Add(new ToolProperty(
+                    toolAttribute.PropertyName,
+                    toolAttribute.PropertyType,
+                    toolAttribute.Description,
+                    toolAttribute.Required));
                 continue;
             }
 
-            properties.Add(new ToolProperty(toolAttribute.PropertyName, toolAttribute.PropertyType, toolAttribute.Description, toolAttribute.Required));
+            var triggerAttribute = parameter.GetCustomAttribute<McpToolTriggerAttribute>();
+            if (triggerAttribute is not null && IsPoco(parameter.ParameterType) && parameter.ParameterType != typeof(ToolInvocationContext))
```

Comment:
@fabiocav wanted to get initial feedback before I continue with this work. This PR works E2E, with POCO and non-POCO bindings. But I am wondering if this is the right approach and/or if I have missed anything with the approach I am taking here.